### PR TITLE
Improved new sample group UI, performance boost

### PR DESCRIPTION
**Pull request by mokolodi1**
This PR should be considered only after #36 has been merged. (#36 adds the HTTP methods that are needed for the data set search box.)

See the commit comments here for more: https://github.com/UCSC-MedBook/MedBook_/commit/687a3b7d40055d84be3b99e37a10a4beb623895e

### CODE REVIEW

File: patient-care/webapp/client/shared/editSampleGroup.js [hunk 1]
```diff
@@ -51,6 +57,46 @@ Template.editSampleGroup.onRendered(function () {
...
     position : "bottom right",
+    content: "Sample group version",
+  });
+
+  instance.$(".data-set-search").search({
+    apiSettings: {
+      url: `${location.origin}/search/data-sets` +
+          `?token=${Accounts._storedLoginToken()}&q={query}`,
+      onResponse(response) {
+        // modify the response to remove existing data sets
```
Comment (e-t-k) [inline patient-care/webapp/client/shared/editSampleGroup.js:68]: When a data set is added to the sample group & then removed again, the response doesn't refresh, so you can't see the data set in the "add a data set list" until you change what string you are searching for.

File: patient-care/webapp/client/manageObjects/manageObjects.js [hunk 2]
```diff
@@ -124,6 +124,68 @@ Template.manageObjectsGrid.onCreated(function () {
...
+  // keep track of whether it's been stickied yet
+  // start this out as false because it hasn't been stickied yet
+  let stickied = false;
+
+  // Make the manage object detail sticky so it moves down
+  // the page as the user scrolls down.
+  function refreshSticky() {
+    // reactively watch the count of the items on the left so it'll update
+    // when it changes, but don't do anything with the return value
+    let hi = getObjects(instance).count();
```
Comment (e-t-k) [inline patient-care/webapp/client/manageObjects/manageObjects.js]: I'm confident there's a better way to depend reactively on a variable, but I don't know enough meteor to tell you what that is.

File: patient-care/webapp/client/manageObjects/manageObjects.js [hunk 3]
```diff
@@ -124,6 +124,68 @@ Template.manageObjectsGrid.onCreated(function () {
...
+  deferredRefreshSticky = () => {
+
+  };
+
+  // call refreshSticky when a reactive variable changes
+  instance.autorun(() => {
+    refreshSticky();
+  });
+
+  // only call it 300 seconds after the thing has stopped resizing
```
Comment (e-t-k) [inline patient-care/webapp/client/manageObjects/manageObjects.js]: 300 milliseconds, surely

**Review by e-t-k — APPROVED** [review]
Everything looks acceptable! I'm happy for this to be merged to master.

File: patient-care/webapp/client/shared/editSampleGroup.js (hunk 1 above)
Comment (mokolodi1) [reply patient-care/webapp/client/shared/editSampleGroup.js:68]: This will be fixed because I'm changing it back to a `.ui.dropdown`.

Comment (e-t-k) [reply patient-care/webapp/client/shared/editSampleGroup.js:68]: In a later pull, presumably. This is annoying but shouldn't block this pull from going to prod.

File: patient-care/webapp/client/manageObjects/manageObjects.js (hunk 2 above)
Comment (mokolodi1) [reply patient-care/webapp/client/manageObjects/manageObjects.js]: I forgot to remove the dummy variable `hi`, but I don't know of a better way to do it. You're correct that this `getObjects` call should arguably be in the `Tracker.autorun` so it isn't called when it doesn't need to be.

File: patient-care/webapp/client/shared/editSampleGroup.js (hunk 1 above)
Comment (mokolodi1) [reply patient-care/webapp/client/shared/editSampleGroup.js:68]: Yes! It's not committed yet.

**Pull request merged**